src/catch2/reporters/catch_reporter_listening.cpp

```diff
@@ -15,151 +17,157 @@ namespace Catch {
         m_listeners.push_back( std::move( listener ) );
     }
 
-    void ListeningReporter::addReporter(IStreamingReporterPtr&& reporter) {
-        assert(!m_reporter && "Listening reporter can wrap only 1 real reporter");
-        m_reporter = std::move( reporter );
-        m_preferences.shouldRedirectStdOut = m_reporter->getPreferences().shouldRedirectStdOut;
+    void ListeningReporter::addReporter( IStreamingReporterPtr&& reporter ) {
+        m_preferences.shouldRedirectStdOut |= reporter->getPreferences().shouldRedirectStdOut;
+
+        // We will need to output the captured stdout if there are reporters
+        // that do not want it captured.
+        m_haveNoncapturingListeners |= !reporter->getPreferences().shouldRedirectStdOut;
+
+        m_preferences.shouldReportAllAssertions |=
+            reporter->getPreferences().shouldReportAllAssertions;
+
+        m_listeners.push_back( std::move( reporter ) );
     }
 
     void ListeningReporter::noMatchingTestCases( std::string const& spec ) {
         for ( auto const& listener : m_listeners ) {
             listener->noMatchingTestCases( spec );
         }
-        m_reporter->noMatchingTestCases( spec );
```

Comment:
I see the idea under unifying listeners and reporters, but we promise that listeners are always called before reporters. We cannot promise this unless we require that all calls to `addListener` are done before `addReporter` calls.

Reply:
Can the registered listeners/reporters change mid-operation? I.e., can we get away with just "sorting" the list at some point? I'd rather avoid the code duplication if possible, unless of course it remains the easiest way.

Reply:
Adding more listeners/reporters after startup is not supported, once we add all registered listeners and selected reporters, that's the final set.

Reply:
However I would like to avoid having some sort of `Multireporter::finalize` member function, so maybe it should internally keep track of "last {reporter/listener} insert index" and just insert new reporters/listeners to the right place.

There shouldn't ever be enough reporters/listeners that the potential n^2 performance comes at a meaningful runtime cost.